plugins/scaffolder/src/next/TemplateWizardPage/TemplateWizardPage.test.tsx

```diff
@@ -103,18 +103,30 @@ describe('TemplateWizardPage', () => {
       fireEvent.click(await findByRole('button', { name: 'Create' }));
     });
 
-    // The "Next Step" button should have fired an event
+    // The "Next Step" button should have fired few events
     expect(analyticsMock.getEvents()[0]).toMatchObject({
       action: 'click',
       subject: 'Next Step (1)',
       context: { entityRef: 'template:default/test' },
     });
 
-    // And the "Create" button should have fired an event
     expect(analyticsMock.getEvents()[1]).toMatchObject({
+      action: 'click',
+      subject: '[template:default/test]: Next Step (1)',
```

Comment:
Why do we need the `entityRef` in the `subject` if it's already in the `context`? Same for below too.

Reply:
@benjdlambert It's related to what is sent to analytics.

It comes down to this:
https://github.com/backstage/backstage/blob/master/plugins/analytics-module-ga4/src/apis/implementations/AnalyticsApi/GoogleAnalytics4.ts#L201

Context information is not submitted

Reply:
Sorry, just catching up on a load of things.

It [should be](https://github.com/backstage/backstage/blob/master/plugins/analytics-module-ga4/src/apis/implementations/AnalyticsApi/GoogleAnalytics4.ts#L168) though, or at least configurable which context keys to submit?

Feels a little unfortunate to duplicate this data, especially without structured data to represent it. Might make querying the data pretty difficult?

Reply:
That's what I see in practice now in GA4

![Screenshot 2024-04-11 at 23 54 30](https://github.com/backstage/backstage/assets/2265094/7c67d86a-b197-438f-bc5a-3dbf41aedb3e)

We can proceed with a configuration, how would it might look like?

Reply:
I wonder if you can pass in `allowedContexts: ['*']` to the analytics constructor and see what the context looks like and if it's enough? https://github.com/backstage/backstage/blob/master/plugins/analytics-module-ga4/src/apis/implementations/AnalyticsApi/GoogleAnalytics4.ts#L58C5-L58C20

Reply:
Sorry only noticed your comment

Actually it's already on:

```
app
  analytics:
    ga4:
      measurementId: XXXX
      testMode: false
      debug: false
      allowedContexts: ['*']
      allowedAttributes: ['*']
      contentGrouping: 'pluginId'
```

Reply:
@acierto coming back round to this again :tada:

I wonder if you need to setup custom dimensions in GA to be able to see the context values. https://support.google.com/analytics/answer/14239696?hl=en#zippy=%2Canalyze-the-custom-dimension-in-a-report%2Canalyze-the-custom-dimension-in-an-exploration%2Cbuild-an-audience-using-the-dimension

Thinking that this should be enough, not sure that it's good practice to overload the label with the `entityRef`.  It makes querying this hard I would think, especially in other implementations of analytics.

Reply:
@benjdlambert Thanks for coming back to this. 

We have dimensions configured, but maybe we have to add also a context there.
Let me check with someone who has permissions to do it, seems like I have only read-only access 😅

Reply:
Now it appears. Not the best thing to work with dimensions in GA4, you can choose only 1: label, subject or context, not all together.

I'm closing this PR, and created a new one https://github.com/backstage/backstage/pull/24537 

Cause I can't merge this PR successfully with master